libs/api/auth/feature/src/lib/api-auth-strategy-telegram.controller.ts

```diff
@@ -1,97 +1,26 @@
-import { Controller, Post, Body, UseGuards, Req, Res } from '@nestjs/common'
-import { objectToAuthDataMap, AuthDataValidator } from "@telegram-auth/server"
-import { IdentityProvider } from '@prisma/client'
-import { ApiAnonJwtGuard } from '@pubkey-link/api-auth-data-access'
-import { ApiAuthRequest } from '@pubkey-link/api-auth-data-access'
-import { ApiAuthStrategyService } from '@pubkey-link/api-auth-data-access'
-import { ApiAuthService } from '@pubkey-link/api-auth-data-access'
+import { Body, Controller, Post, Req, Res, UseGuards } from '@nestjs/common'
+import {
+  ApiAnonJwtGuard,
+  ApiAuthJwtGuard,
+  ApiAuthRequest,
+  ApiAuthStrategyTelegramService,
+  TelegramAuthPayload,
+} from '@pubkey-link/api-auth-data-access'
 import { Response } from 'express-serve-static-core'
 
-interface TelegramAuthPayload {
-  id: number
-  first_name: string
-  last_name?: string
-  username?: string
-  photo_url?: string
-  auth_date: number
-  hash: string
-}
-
 @Controller('auth/telegram')
 export class ApiAuthStrategyTelegramController {
-  private validator: AuthDataValidator
-  constructor(private readonly authStrategy: ApiAuthStrategyService, private readonly service: ApiAuthService) {
-    this.validator = new AuthDataValidator({
-      botToken: process.env['TELEGRAM_BOT_TOKEN'] as string,
-    })
-  }
-
-  @Post('validate')
-  @UseGuards(ApiAnonJwtGuard)
-  async validateTelegram(@Body() data: TelegramAuthPayload, @Req() req: ApiAuthRequest) {
-    try {
-      const authData = objectToAuthDataMap(JSON.parse(JSON.stringify(data)))
-      const user = await this.validator.validate(authData)
+  constructor(private readonly service: ApiAuthStrategyTelegramService) {}
```

Comment:
The controller is now only responsible for accepting the payload, handing authentication, and pass that to the service.